src/NuGet.Core/NuGet.Common/PathUtil/PathUtility.cs

```diff
@@ -490,7 +490,7 @@ public static string StripLeadingDirectorySeparators(string filename)
             filename = GetPathWithForwardSlashes(filename);
             var currentDirectoryPath = $"./";
 
-            if (filename.StartsWith(currentDirectoryPath))
+            if (filename.StartsWith(currentDirectoryPath, StringComparison.OrdinalIgnoreCase))
```

Comment:
From @nkolev92 [comment](https://github.com/NuGet/NuGet.Client/pull/4888#discussion_r1012330062), we should make this comparison file-system aware:

```suggestion
            if (filename.StartsWith(currentDirectoryPath, PathUtil.GetStringComparisonBasedOnOS()))
```